src/management/TodoContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useReducer } from 'react';
+import { State } from '../types/State';
+import { Action, reducer } from './reducer';
+import { Filter } from '../types/Filter';
+import { USER_ID, getTodos } from '../api/todos';
+
+const initialState: State = {
+  todos: [],
+  filterBy: Filter.all,
+  errorMessage: '',
+};
+
+export const StateContext = React.createContext(initialState);
+export const DispatchContext
+  = React.createContext<React.Dispatch<Action>>(() => { });
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const GlobalStateProvider: React.FC<Props> = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const userId = USER_ID;
+
+  useEffect(() => {
+    dispatch({ type: 'errorMessage', payload: '' });
+
+    getTodos(userId)
+      .then(todosFromServer => {
+        const todos = todosFromServer.map(todo => ({
```

Comment:
you map todos to get rid of `createdAt` and `updatedAt`? It's not necessary (FYI)
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/86923089/897d071e-eaca-45f7-b5f7-3f80ad27c5cf)